src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentGeneratedOutputTracker.cs

```diff
@@ -86,13 +96,77 @@ public DocumentGeneratedOutputTracker Fork()
             // Drop reference so it can be GC'ed
             _older = null;
 
-
-            // Cache the tag helpers so the next version can use them
+            // Cache the tag helpers and imports so the next version can use them
             _tagHelpers = tagHelpers;
+            _imports = imports;
+
+            var importSources = new List<RazorSourceDocument>();
+            foreach (var item in imports)
+            {
+                var sourceDocument = await GetRazorSourceDocumentAsync(item.Import);
+                importSources.Add(sourceDocument);
+            }
+
+            var documentSource = await GetRazorSourceDocumentAsync(document);
 
             var projectEngine = project.GetProjectEngine();
-            var projectItem = projectEngine.FileSystem.GetItem(document.FilePath);
-            return projectItem == null ? null : projectEngine.ProcessDesignTime(projectItem);
+
+            return projectEngine.ProcessDesignTime(documentSource, importSources, tagHelpers);
+        }
+
+        private async Task<RazorSourceDocument> GetRazorSourceDocumentAsync(DocumentSnapshot document)
+        {
+            var sourceText = await document.GetTextAsync();
+
+            return sourceText.GetRazorSourceDocument(document.FilePath);
+        }
+
+        private async Task<IReadOnlyList<ImportItem>> GetImportsAsync(ProjectSnapshot project, DocumentSnapshot document)
+        {
+            var imports = new List<ImportItem>();
+            foreach (var snapshot in document.GetImports())
+            {
+                var versionStamp = await snapshot.GetTextVersionAsync();
+                imports.Add(new ImportItem(snapshot.FilePath, versionStamp, snapshot));
```

Comment:
Is it possible to avoid interacting on to 'snapshots' at this level of the system? I don't want to see the external API leak too far into the the implementation.